modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/persistence/Serializer.scala

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.bluebrain.nexus.kg.core.persistence
+
+import java.io.NotSerializableException
+import java.nio.charset.StandardCharsets.UTF_8
+
+import akka.serialization.SerializerWithStringManifest
+import ch.epfl.bluebrain.nexus.commons.http.JsonOps._
+import ch.epfl.bluebrain.nexus.kg.core.resources.{Event, Key}
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.auto._
+import io.circe.java8.time._
+import io.circe.generic.extras.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.parser.decode
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder, Json}
+
+/**
+  * Akka ''SerializerWithStringManifest'' class definition for all events.
+  * The serializer provides the types available for serialization.
+  */
+object Serializer {
+
+  private implicit val config: Configuration = Configuration.default.withDiscriminator("type")
+  private final val EventName                = "event"
+
+  private implicit val eventEncoder: Encoder[Event] = {
+    val enc = deriveEncoder[Event]
+    Encoder.instance { ev =>
+      enc(ev).removeKeys("id") deepMerge ev.id.asJson
+    }
+  }
+
+  private implicit val eventDecoder: Decoder[Event] = {
+    val dec = deriveDecoder[Event]
+    Decoder.instance { hc =>
+      val json = hc.value
+      for {
+        key <- json.as[Key]
+        combined = json deepMerge Json.obj("id" -> key.asJson)
+        event <- dec(combined.hcursor)
+      } yield event
+    }
+  }
+
+  class EventSerializer extends SerializerWithStringManifest {
+    override final def manifest(o: AnyRef): String =
+      o match {
+        case _: Event => EventName
+        case _        => throw new IllegalArgumentException(s"Unable to generate manifest for '$o'")
+      }
+
+    override final def toBinary(o: AnyRef): Array[Byte] =
+      o match {
+        case ev: Event => ev.asJson.noSpaces.getBytes(UTF_8)
+        case _         => throw new IllegalArgumentException(s"Unable to encode to binary; unknown type '$o'")
+
+      }
+
+    override final def fromBinary(bytes: Array[Byte], manifest: String): AnyRef =
+      manifest match {
+        case `EventName` =>
+          decode[Event](new String(bytes, UTF_8)) match {
+            case Left(err)    => throw err
+            case Right(value) => value
+          }
+        case _ => throw new NotSerializableException(manifest)
+      }
+
+    val identifier = 1234567
```

Comment:
Is this OK? :smile:

Reply:
NO! :)

How about `1050` (`"nexus-json".getBytes.map(_.toInt).sum`) ?

Reply:
Yes: https://doc.akka.io/docs/akka/2.5.4/java/serialization.html

Reply:
it doesn't mean we should use `1234567`...